src/main/java/bio/terra/workspace/service/job/JobService.java

```diff
@@ -123,12 +136,27 @@ protected String submit(
 
   void waitForJob(String jobId) {
     try {
-      stairway.waitForFlight(jobId, 10, appConfig.getStairwayTimeoutSeconds() / 10);
-    } catch (StairwayException stairwayEx) {
+      waitForFlight(
+          jobId,
+          appConfig.getStairwayPollingIntervalSeconds(),
+          appConfig.getStairwayTimeoutSeconds() / appConfig.getStairwayPollingIntervalSeconds());
+    } catch (StairwayException | InterruptedException stairwayEx) {
       throw new InternalStairwayException(stairwayEx);
     }
   }
 
+  private FlightState waitForFlight(String flightId, int pollSeconds, int pollCycles)
+      throws DatabaseOperationException, FlightException, InterruptedException {
+    for (int pollCount = 0; pollCount < pollCycles; ++pollCount) {
+      TimeUnit.SECONDS.sleep(pollSeconds);
+      FlightState state = stairway.getFlightState(flightId);
+      if (!state.isActive()) {
+        return state;
+      }
+    }
+    throw new FlightException("Flight did not complete in the allowed wait time");
+  }
```

Comment:
could you rewrite this to use `ScheduledExecutorService` instead of `sleep`?

Reply:
Done. I'm not familiar with `ScheduledExecutorService` so this may not be what you had in mind - let me know if I've messed something up.

Reply:
this looks right to me! I originally thought we could use `scheduleWithFixedDelay` but that doesn't return a value; I think you've done the right thing here.